frontend/src/metabase/core/components/Button/Button.tsx

```diff
@@ -37,7 +37,7 @@ interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
   as?: ElementType;
   className?: string;
 
-  icon?: string;
+  icon?: string | ReactNode;
```

Comment:
I think being able to pass the whole icon element is idiomatic as we're hacking it now by rendering icons in a `<ButtonTextContainer />`

and the problem with icon not staying center is we need to make `ButtonTextContainer` a flex container and I think that could affect a lot of components that use `Button` in a bad way. So this is more smaller scoped and displays a custom icon where it's supposed to be displayed.

#### With this change
![image](https://user-images.githubusercontent.com/1937582/182381738-4b895565-1630-44ae-bae7-f5cac5ca3fd7.png)

#### Before this change (icon renders in a text container)
![image](https://user-images.githubusercontent.com/1937582/182381876-f0e3cfb1-bb16-4345-a01a-b1348771514d.png)